From e513edbcb6d870c9ce956f0a092d01540073b750 Mon Sep 17 00:00:00 2001
From: Grigory Kletsko <grigory.kletsko@cogentembedded.com>
Date: Tue, 13 Dec 2016 00:43:11 +0300
Subject: [PATCH] Add widget_set_surface_allocation func

---
 clients/toytoolkit.h |  3 +++
 clients/window.c     | 12 ++++++++++++
 2 files changed, 15 insertions(+)

diff --git a/clients/toytoolkit.h b/clients/toytoolkit.h
index 67c2720..45d322a 100644
--- a/clients/toytoolkit.h
+++ b/clients/toytoolkit.h
@@ -532,6 +532,9 @@ void
 widget_set_allocation(struct widget *widget,
 		      int32_t x, int32_t y, int32_t width, int32_t height);
 void
+widget_set_surface_allocation(struct widget *widget,
+		      int32_t x, int32_t y, int32_t width, int32_t height);
+void
 widget_set_size(struct widget *widget, int32_t width, int32_t height);
 void
 widget_set_transparent(struct widget *widget, int transparent);
diff --git a/clients/window.c b/clients/window.c
index 3b7def3..c0b25e8 100644
--- a/clients/window.c
+++ b/clients/window.c
@@ -1761,6 +1761,18 @@ widget_set_allocation(struct widget *widget,
 }
 
 void
+widget_set_surface_allocation(struct widget *widget,
+		      int32_t x, int32_t y, int32_t width, int32_t height)
+{
+	if (widget->surface) {
+		widget->surface->allocation.x = x;
+		widget->surface->allocation.y = y;
+		widget->surface->allocation.width = width;
+		widget->surface->allocation.height = height;
+	}
+}
+
+void
 widget_set_transparent(struct widget *widget, int transparent)
 {
 	widget->opaque = !transparent;
-- 
2.7.4

